### PR TITLE
Test: add community tests

### PR DESCRIPTION
**Pull request by mhenriette**
# Submit a pull request

- [x] This is not a duplicate of an existing pull request.
- [x] No existing features have been broken without good reason.
- [x] Your commit messages are detailed
- [ ] The code style guidelines have been followed.
- [ ] Documentation has been updated to reflect your changes.
- [x] Tests have been added or updated to reflect your changes.
- [x] All tests pass.

---

## Pull Request Details
- add tests for community 
- create an absolute path for mocks 
:question:

## Breaking Changes

none

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

### CODE REVIEW

File: src/components/sections/communities/_partials/Section.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface SectionProps {
  */
 export default function Section({ title, titleBold = true, subtitle, id, hideSubtitleOnMobile = false, children, className }: SectionProps): ReactElement {
   return (
-    <div id={id} className={`text-xl md:text-.5xl px-0 py-6 ${className}`}>
+    <div data-testid="sectionId" id={id} className={`text-xl md:text-.5xl px-0 py-6 ${className}`}>
```
Comment (Samantha-KY) [inline src/components/sections/communities/_partials/Section.tsx:38]: can you keep the consistency and add an argument for this testid
```suggestion
    <div data-testid="sectionId" id={id} className={`text-xl md:text-.5xl px-0 py-6 ${className}`}>
```

File: __tests__/components/sections/communities/overview/scoreboard/index.test.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    expect(screen.getByTestId("scoreboardId")).toBeInTheDocument();
+  });
+
+  it("renders scoreboard cards when not loading and list is not empty", () => {
+    renderWithRedux(<ScoreboardOverview />, {
+      scoreboard: { list: [mockScoreboard], loading: false, filterBy: "" },
+    });
+    if ([mockScoreboard] && [mockScoreboard].length !== 0) {
+      [mockScoreboard].slice(0, items).forEach((item) => {
+        expect(screen.getByText(item.user.displayName)).toBeInTheDocument();
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/scoreboard/index.test.tsx:32]:
```suggestion
        expect(screen.getByText(item.user.displayName)).toBeInTheDocument();
```
you can add another test to check if the text is as expected

File: __tests__/components/sections/communities/overview/scoreboard/_partials/FilterOption.test.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import FilterOption from "@/components/sections/communities/overview/scoreboard/_partials/FilterOption";
+
+describe("FilterOption", () => {
+  it("renders with label and is not checked by default", () => {
+    const label = "Test Label";
+    render(<FilterOption label={label} value="1" data="2" />);
+
+    expect(screen.getByLabelText(label)).toBeInTheDocument();
+    expect(screen.getByRole("radio")).not.toBeChecked();
+  });
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/scoreboard/_partials/FilterOption.test.tsx:12]: check if the text rendered is "Test Label"

File: __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+jest.mock("next/router", () => ({
+  useRouter: jest.fn(),
+}));
+
+describe("Filter", () => {
+  const mockRouter = {
+    query: { slug: "test-slug" },
+    locale: "en",
+    push: jest.fn(),
+  };
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx:16]: You can add this at the top in the useRouter you mocked

File: __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+  beforeEach(() => {
+    (useRouter as jest.Mock).mockReturnValue(mockRouter);
+  });
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx:20]: There is no need for this if you add the content of mockRouter in the jest.mock on top

File: __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+  it("renders filter options", () => {
+    renderWithRedux(<Filters />);
+    expect(screen.getByText("Filter by")).toBeInTheDocument();
+    filterOptions.forEach((option) => {
+      expect(screen.getByText(option.label)).toBeInTheDocument();
+      expect(screen.getByDisplayValue(option.value)).toBeInTheDocument();
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx:27]: Check if the values in the document are the expected ones

File: __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      expect(value.getAttribute("value")).toContain("new filter option");
+    });
+  });
+
+  it("renders sort options", () => {
+    renderWithRedux(<Filters />);
+    expect(screen.getByText("Sort")).toBeInTheDocument();
+    sortingOptions.forEach((option) => {
+      expect(screen.getByText(option.label)).toBeInTheDocument();
+      expect(screen.getByDisplayValue(option.value)).toBeInTheDocument();
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/scoreboard/Filter.test.tsx:45]: Same here, check if the text in the document are the expected ones

File: __tests__/components/sections/communities/overview/Courses.test.tsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+describe("CoursesOverview", () => {
+  it("renders the courses overview section with course cards", () => {
+    renderWithRedux(<CoursesOverview />, {
+      courses: { current: mockCourse, list: [mockCourse], content: mockCourse.community, count: 1, menus: [] },
+      community: { current: mockCommunity, list: [mockCommunity], courses: [mockCourse], status: "succeeded", error: "" },
+    });
+    expect(screen.getByText("communities.overview.courses.title")).toBeInTheDocument();
+    [mockCourse].forEach((course) => {
+      expect(screen.getByText(course.name)).toBeInTheDocument();
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/Courses.test.tsx:16]: Check if the text displayed is the expected

**Review by Samantha-KY — CHANGES_REQUESTED** [review]

File: __tests__/components/sections/communities/overview/scoreboard/_partials/FilterOption.test.tsx (hunk 3 above)
Comment (mhenriette) [reply __tests__/components/sections/communities/overview/scoreboard/_partials/FilterOption.test.tsx:12]: it is already tested, when checking if label is in the document.

**Review by mbukeRepo — COMMENTED** [review]
Recommendation: Avoid Hardcoding Data for Testing; Use Faker.js Instead

For more robust and flexible testing, it is recommended to avoid hardcoding data. Instead, consider using Faker.js. This library allows you to generate realistic and varied test data dynamically, which can enhance the reliability and thoroughness of your testing processes. By utilizing Faker.js, you can simulate a wide range of scenarios and edge cases, ensuring that your code is well-tested and resilient to different inputs. This practice not only improves test coverage but also enhances the maintainability of your test suite.

https://www.npmjs.com/package/@faker-js/faker

File: __tests__/components/sections/communities/overview/Courses.test.tsx [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  it("renders the courses overview section with course cards", () => {
+    renderWithRedux(<CoursesOverview />, {
+      courses: { current: mockCourse, list: [mockCourse], content: mockCourse.community, count: 1, menus: [] },
+      community: { current: mockCommunity, list: [mockCommunity], courses: [mockCourse], status: "succeeded", error: "" },
+    });
+    expect(screen.getByText("communities.overview.courses.title")).toBeInTheDocument();
+    [mockCourse].forEach((course) => {
+      const courseElement = screen.getByText(course.name);
+      expect(courseElement).toBeInTheDocument();
+      expect(courseElement.textContent).toBe(course.name);
```
Comment (Samantha-KY) [inline __tests__/components/sections/communities/overview/Courses.test.tsx:18]: can you check what the text is if is as expected

**Review by Samantha-KY — APPROVED** [review]

File: __tests__/components/sections/communities/_partials/Header.test.tsx
Comment (serapieTuyishime) [inline __tests__/components/sections/communities/_partials/Header.test.tsx:1]: could you use renderWithRedux?

File: __tests__/components/sections/communities/overview/MainHeader.test.tsx [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  };
+  it("renders community section component", () => {
+    renderWithRedux(<CommunitySection />, communityMockState);
+    expect(screen.getByRole("heading", {name:mockCommunity.name})).toBeInTheDocument();
+    const summaryElements = screen.queryAllByText(mockCommunity.summary);
+    summaryElements.forEach(element=>{
+      expect(element).toBeInTheDocument()
+    })
+    if(mockCommunity.icon){
+      expect(screen.getByRole('img')).toBeInTheDocument()
```
Comment (serapieTuyishime) [inline __tests__/components/sections/communities/overview/MainHeader.test.tsx:21]: IMO, i think that this will get any image and in this case we are only testing for community icon only.

File: __tests__/components/sections/communities/overview/Wrapper.test.tsx [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+jest.mock("next/router", () => ({
+  useRouter: () => ({
+    asPath: "next",
+  }),
+}));
+
+
+describe("Wrapper", () => {
+  it("displays the wrapper", () => {
```
Comment (serapieTuyishime) [inline __tests__/components/sections/communities/overview/Wrapper.test.tsx:14]:
```suggestion
  it("displays the wrapper without all passed children components", () => {
```

File: __tests__/components/sections/communities/overview/_partials/SectionWrapper.test.tsx [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+      <SectionWrapper title={title} description={description}>
+        <div>Test Children</div>
+      </SectionWrapper>
+    );
+    expect(screen.getByText(title)).toBeInTheDocument();
+    expect(screen.getByText(description)).toBeInTheDocument();
+    expect(screen.getByText("Test Children")).toBeInTheDocument();
+  });
+
+  it("renders without title and description", () => {
```
Comment (serapieTuyishime) [inline __tests__/components/sections/communities/overview/_partials/SectionWrapper.test.tsx:19]:
```suggestion
  it("renders without title and description when necessary", () => {
```

or something closely related to it

**Review by serapieTuyishime — CHANGES_REQUESTED** [review]

File: __tests__/components/sections/communities/overview/MainHeader.test.tsx (hunk 10 above)
Comment (mhenriette) [reply __tests__/components/sections/communities/overview/MainHeader.test.tsx:21]: There's only one image in the component, so the only image in the document is for the icon, otherwise, if we had more than one image we would get an error.
but we can get the image via its alt name as well.

**Review by serapieTuyishime — APPROVED** [review]
